app/src/main/java/waiter/EchoServer.java

```diff
@@ -0,0 +1,20 @@
+package waiter;
+
+import waiter.Communicator.Reporter;
+
+import java.io.IOException;
+import java.net.SocketException;
+
+public record EchoServer(Reporter reporter) {
+
+    public void start() {
+
+        try {
+            reporter.communicate();
+        } catch (SocketException exception) {
+            System.out.println("Sorry, connection could not be established or has been broken, please try running the server and connecting again");
+        } catch (IOException exception) {
+            System.out.println("Sorry, there was a problem with your input, please try running the server and connecting again");
```

Comment:
That is a good point that I didn't think of and makes a lot of sense to me now that you mention it, how does `"Sorry, there was a problem with reading or writing your message, please try running the server and connecting again"` sound? or do you think it would be better (because its more general) to just say ".... problem with your message...." instead of "reading or writing"?

Reply:
Maybe something like "... an error occurred when sending/receiving a message, please try again later..."